updater/src/Updater/Generate/Template.purs

```diff
@@ -154,75 +146,90 @@ runTemplate opts (Template { from, to }) = do
 
   FS.Extra.writeTextFile to (replaceVariables opts.variables templateContents)
 
+-- | Keep JS only templates when using JS.
+filterByType :: Boolean -> Array TemplateSource -> Array TemplateSource
+filterByType true templates = templates
+filterByType false templates = filter (not <<< (JS == _) <<< sourceType) templates
+
+-- | Define the source and destination of a given template by its type and
+-- | source path. Common templates default to _base_ unless using JS.
+templateFromSource :: Boolean -> TemplateSource -> Template
```

Comment:
Since the `TemplateSource` type already contains the `to` path, it may be more ergonomic to just have a function which applies the correct prefix (`js` or `base`) to that path. What do you think?